src/types.jl

```diff
@@ -777,7 +777,14 @@ function show_ref(io, f, args)
 end
 
 function show_call(io, f, args)
-    fname = istree(f) ? Symbol(repr(f)) : nameof(f)
+    fname = if istree(f)
+        Symbol(repr(f))
+    elseif f isa Function
+        nameof(f)
+    else
+        # Dummy for callable structs
+        Symbol()
```

Comment:
How does this actually show the name of the struct type? This should be also `Symbol(repr(f))` or something that gets the type name as a symbol in my opinion.